Added Bank and BankUnitTests

todorzag opened the pull request.

todorzag committed:
Added Bank and BankUnitTests

todorzag committed:
Remade some unit test to be more readable

mvlt commented on Bank/BankAccount/BankAccount.cs:
```diff
@@ -0,0 +1,135 @@
+                    ? transaction.Amount * -1
+                    : transaction.Amount;
+
+                accountHistory.AppendLine($"A {transaction.Type} was made on: {transaction.Date}");
```
Why do we need 3 AppendLines?

mvlt commented on Bank/BankAccount/Program.cs:
```diff
@@ -0,0 +1,98 @@
+            Console.WriteLine("Make another transaction?");
+            string answer = Console.ReadLine();
+
+            switch (answer)
```
Why do we have switch statement with 1 case?

mvlt commented on Bank/BankAccount/Program.cs:
```diff
@@ -0,0 +1,98 @@
+            return note;
+        }
+
+        private static string ValidateTransactionType(string transactionType)
```
Not doing what the name of the function is saying.

mvlt commented on Bank/BankAccount/Transaction.cs:
```diff
@@ -0,0 +1,68 @@
+            }
+        }
+
+        private void ValidateTypeThrowException(string type)
```
Are we validating this twice?

mvlt reviewed (changes requested):
Check method's and field's order priority

todorzag committed:
Changes made:
 - Rearranged BankAccount methods and fields
 - Removed unnecessary validate methods
 - Added default to TransactionHandler to throw Exception
 - Removed Transaction type test

todorzag committed:
Added Type property to Transaction to fix error
